integration-libs/cdc/root/service/cdc-js.service.ts

```diff
@@ -365,6 +435,142 @@ export class CdcJsService implements OnDestroy {
     }
   }
 
+  /**
+   * Trigger CDC user email update.
+   *
+   * @param password
+   * @param newEmail
+   */
+  updateUserEmailWithoutScreenSet(
+    password: string,
+    newEmail: string
+  ): Observable<{ status: string }> {
+    if (
+      !password ||
+      password?.length === 0 ||
+      !newEmail ||
+      newEmail?.length === 0
+    ) {
+      return throwError(null);
+    } else {
+      //Verify the password by attempting to login
+      let email = this.getLoggedInUserEmail();
+      return this.loginUserWithoutScreenSet(
+        email,
+        password,
+        'RESET_EMAIL'
```

Comment:
Let's elaborate more in the code comment, _why_ we implemented it such way:

```suggestion
      // Verify the password by attempting to login
      // - CDC doesn't require to verify password before changing an email, but the default Spartacus requires it.
      // - CDC doesn't have any specific api, for verifying a password, so as a _workaround_ we call the login API of CDC. We pass a special `context` parameter `'RESET_EMAIL'`, which doens't mean instruct the CDC API to reset email, but only conditionally avoid the full CDC login flow. Instead we want only half of the CDC login flow, just to verify if the password was correct.
      let email = this.getLoggedInUserEmail();
      return this.loginUserWithoutScreenSet(
        email,
        password,
        'RESET_EMAIL'
```

Reply:
thanks @Platonn